spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -300,6 +306,93 @@ public void afterSingletonsInstantiated() {
 
 	}
 
+	private void registerProducer(MessageProducer messageProducer) {
```

Comment:
Should be below the next method.

Reply:
It is called from the `afterSingletonsInstantiated()` first time, so this is normal order for the private methods.

Reply:
But it's also called from `postProcessAfterInitialization()`. I believe the rule is called methods should always be below the last place that calls it.

Reply:
Fine with me, but I've followed this rule so far: https://github.com/spring-projects/spring-framework/wiki/Code-Style#java-source-file-organization

> Note that private or protected methods called from method implementations should be placed immediately below the methods where they're used. In other words if there 3 interface method implementations with 3 private methods (one used from each), then the order of methods should include 1 interface and 1 private method in sequence, not 3 interface and then 3 private methods at the bottom.

Reply:
Maybe it's just my interpretation then; I guess those guidelines don't specifically address this use case.

Reply:
OK. Fine with me.
Will fix. 😄 

Not sure though, what happened to the `org.springframework.integration.monitor.MessageSourceTests` on Travis...